server/sandbox_network.go

```diff
@@ -20,8 +20,11 @@ import (
 // or an error
 func (s *Server) networkStart(ctx context.Context, sb *sandbox.Sandbox) (podIPs []string, result cnitypes.Result, retErr error) {
 	overallStart := time.Now()
-	// give a network Start call 2 minutes, half of a RunPodSandbox request timeout limit
-	startCtx, startCancel := context.WithTimeout(ctx, 2*time.Minute)
+	// Give a network Start call the entire RunPodSandbox request timeout limit.
+	// This is to prevent the CNI plugin from taking an unbounded amount of time,
+	// but to still allow a long-running sandbox creation to be cached and reused,
+	// rather than failing and recreating it.
+	startCtx, startCancel := context.WithTimeout(ctx, 4*time.Minute)
```

Comment:
why not pass down the main context here if we are using the full timeout?

Reply:
yeah that's a good idea. I am now  actually considering creating a custom timeout independent of the outer context (like: 
`startCtx, startCancel := context.WithTimeout(context.Background(), 6*time.Minute)` ) which would really allow us to work independently of the kubelet's timeout